homeassistant/components/twentemilieu/sensor.py

```diff
@@ -0,0 +1,154 @@
+from homeassistant.core import callback
+from homeassistant.exceptions import PlatformNotReady
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.typing import HomeAssistantType
+
+_LOGGER = logging.getLogger(__name__)
+
+PARALLEL_UPDATES = 1
+
+
+async def async_setup_entry(
+        hass: HomeAssistantType, entry: ConfigEntry, async_add_entities
+) -> None:
+    """Set up Twente Milieu sensor based on a config entry."""
+    twentemilieu = hass.data[DOMAIN][entry.data[CONF_ID]]
+
+    try:
+        await twentemilieu.update()
+    except TwenteMilieuConnectionError as exception:
+        raise PlatformNotReady from exception
+
+    sensors = [
+        TwenteMilieuSensor(
+            twentemilieu,
+            unique_id=entry.data[CONF_ID],
+            name="{} Waste Pickup".format(WASTE_TYPE_NON_RECYCLABLE),
+            waste_type=WASTE_TYPE_NON_RECYCLABLE,
+            icon="mdi:delete-empty",
+        ),
+        TwenteMilieuSensor(
+            twentemilieu,
+            unique_id=entry.data[CONF_ID],
+            name="{} Waste Pickup".format(WASTE_TYPE_ORGANIC),
+            waste_type=WASTE_TYPE_ORGANIC,
+            icon="mdi:delete-empty",
+        ),
+        TwenteMilieuSensor(
+            twentemilieu,
+            unique_id=entry.data[CONF_ID],
+            name="{} Waste Pickup".format(WASTE_TYPE_PAPER),
+            waste_type=WASTE_TYPE_PAPER,
+            icon="mdi:delete-empty",
+        ),
+        TwenteMilieuSensor(
+            twentemilieu,
+            unique_id=entry.data[CONF_ID],
+            name="{} Waste Pickup".format(WASTE_TYPE_PLASTIC),
+            waste_type=WASTE_TYPE_PLASTIC,
+            icon="mdi:delete-empty",
+        ),
+    ]
+
+    async_add_entities(sensors, True)
+
+
+class TwenteMilieuSensor(Entity):
+    """Defines a Twente Milieu sensor."""
+
+    def __init__(
+            self,
+            twentemilieu: TwenteMilieu,
+            unique_id: str,
+            name: str,
+            waste_type: str,
+            icon: str,
+    ) -> None:
+        """Initialize the Twente Milieu entity."""
+        self._available = True
+        self._unique_id = unique_id
+        self._icon = icon
+        self._name = name
+        self._twentemilieu = twentemilieu
+        self._waste_type = waste_type
+        self._unsub_dispatcher = None
+
+        self._state = None
+
+    @property
+    def name(self) -> str:
+        """Return the name of the entity."""
+        return self._name
+
+    @property
+    def icon(self) -> str:
+        """Return the mdi icon of the entity."""
+        return self._icon
+
+    @property
+    def available(self) -> bool:
+        """Return True if entity is available."""
+        return self._available
+
+    @property
+    def unique_id(self) -> str:
+        """Return the unique ID for this sensor."""
+        return "_".join([DOMAIN, self._unique_id, self._waste_type])
```

Comment:
Please use new style string formatting instead.

Reply:
f-strings already?!

Reply:
No, hehe, I call `'{}'.format()` new style. It's not so new anymore.

Reply:
I'll change it, however, I don't conder this formatting of a string and even consider using string formatting for these kinds of things less readable.

Reply:
✅ Addressed. Done.